src/main/java/sk/turn/http/Http.java

```diff
@@ -566,4 +581,22 @@ private Map<String, List<String>> reindexHeaderMap(Map<String, List<String>> hea
 		return newHeaders;
 	}
 
+	private int parseContentLength(String contentLengthString) {
+		return (contentLengthString != null && contentLengthString.matches("^[0-9]+$")) ? Integer.parseInt(contentLengthString) : -1;
+	}
+
+	private static class CopyProgressListener {
```

Comment:
Is this class really necessary? Could be achieved with 1 extra param in copyStream.